Tests/AmazonLocationiOSAuthSDKTests/AuthHelperTests.swift

```diff
@@ -47,41 +45,107 @@ final class AuthHelperTests: XCTestCase {
+    }
+    
+    func testCognitoCredentialsProviderSigner() async throws {
+        let config = readTestConfig()
+        let identityPoolId = config["identityPoolID"]!
+        let region = config["region"]!
+        let placeIndex = config["placeIndex"]!
+        let authHelper = AuthHelper()
+        let authProvider = try? await authHelper.authenticateWithCognitoIdentityPool(identityPoolId: identityPoolId, region: region)
+        let cognitoCredentials = authProvider?.getCognitoProvider()?.getCognitoCredentials()
+        let address = try? await searchPositionAPI(region: region, indexName: placeIndex, position: [40.758023, -73.985564], cognitoCrdentials: cognitoCredentials!)
+        XCTAssertNotNil(address, "Address found")
+    }
+    
+    func searchPositionAPI(region: String, indexName: String, position: [NSNumber], cognitoCrdentials: CognitoCredentials) async throws -> String? {
+        let url = URL(string: "https://places.geo.\(region).amazonaws.com/places/v0/indexes/\(indexName)/search/position")!
+
+        let requestBody: [String: Any] = [
+            "Language": "en",
+            "MaxResults": 10,
+            "Position": position
+        ]
+
+        let requestData: Data
+        do {
+            requestData = try JSONSerialization.data(withJSONObject: requestBody, options: [])
+        } catch {
+            print("Error: Unable to encode request body as JSON")
+            return nil
+        }
+        
+        var headers = HTTPHeaders()
+        headers.add(name: "Content-Type", value: "application/json")
+        
+        let signer = AWSSigner(credentials: cognitoCrdentials, name: "geo", region: region)
+        let signedHeaders = signer.signHeaders(url: url, method: .POST, headers: headers, body: .data(requestData))
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = HTTPMethod.POST.rawValue
+        request.httpBody = requestData
+        
+        for (name, value) in signedHeaders.allHeaders() {
+            request.addValue(value, forHTTPHeaderField: name)
+        }
+        
+        do {
+            let (data, response) = try await URLSession.shared.data(for: request)
+            
+            guard let httpResponse = response as? HTTPURLResponse,
+                  (200...299).contains(httpResponse.statusCode) else {
+                return nil
+            }
+            
+            guard let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
+                  let results = json["Results"] as? [[String: Any]],
+                  let result = results.first,
+                  let properties = result["Place"] as? [String: Any],
+                  let address = properties["Label"] as? String else {
+                return nil
+            }
+            
+            return address
+        } catch {
+            print("Error: \(error.localizedDescription)")
+            throw error
```

Comment:
unnecessary catch

Reply:
This is not unnecessary. Some statements are throwing error in this block